### ISSUE TITLE
Overfit problem with YOLOv8 on CrowdHuman dataset for people recognition

### ISSUE THREAD

**Issue by PietroVignini** [ultralytics/ultralytics#10176]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hi all!!!
I am trying to train YOLOv8 on the CrowdHuman dataset for my master thesis project. 

I made some attempts, changing the optimizer and learning rate and other hyperparameters to try to find the best combination. 
I got an improvement in all metrics (P, R, mAP50, mAP50-95) compared to the pre-trained model on the COCO dataset, but unfortunately in all the attempts I made I noticed box loss, class loss and dfl loss quite worse on the validation set than on the training set, even though they have a similar trend, making me think that the model suffers from overfitting.

**YOLOv8s pretrained on COCO: P = 0.738, R = 0.554, mAP50 = 0.655, mAP50-95 =  0.408**

I add here the parameters I used in the two best attempts and the results:

` results = model.train(model='yolov8s.pt', data=os.path.join(ROOT_DIR, "config.yaml"), epochs=300, patience = 100, 
 batch=32, imgsz=640, device=[0,1], cache=True, save=True, workers=24, pretrained=True, single_cls=True, optimizer='SGD', deterministic=False, cos_lr=True, lr0=1e-2, lrf=1e-2, augment=False, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=15, translate=0.1, scale=0.5, shear=0.0001, perspective=0.0001, fliplr=0.5, mosaic=1.0, mixup=0.0, copy_paste=0.0, verbose=True, visualize=True) `

![results_best2](https://github.com/ultralytics/ultralytics/assets/166723771/2c583523-29fc-41ac-a96a-91ba78ec1801)

` results = model.train(model='yolov8s.pt', data=os.path.join(ROOT_DIR, "config.yaml"), epochs=300, patience = 100, 
 batch=32, imgsz=640, device=[0,1], cache=True, save=True, workers=24, pretrained=True, single_cls=True, optimizer='SGD', deterministic=False, cos_lr=True, lr0=1e-2, lrf=1e-2, augment=True, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=15, translate=0.1, scale=0.5, shear=0.0001, perspective=0.0001,  fliplr=0.5, mosaic=1.0, mixup=0.0, copy_paste=0.0, verbose=True, visualize=True) `

![results_best4](https://github.com/ultralytics/ultralytics/assets/166723771/cf1586b8-2a95-4168-a254-1bcd5e3bbc7d)

In this last attempt, in particular, I enabled augmentation with `augment=True` and noticed that in the last 10 epochs, when mosaic data augmentation is disabled, training losses increase to values closer to the validation dataset losses. Could mosaic augmentation be responsible for the overfitting? Or am I missing something?

Another doubt I have is: what effect does `augment=True` or `augment=False` have? does it actually enable/disable augmentation?

Thank you very much for your assistance.

**Comment by glenn-jocher** [comment]
Hi there! 🌟 It's fantastic that you're using YOLOv8 for your master thesis project on the CrowdHuman dataset. Addressing overfitting is indeed crucial for obtaining a model that performs well not just on the training set but also on unseen data. 

Given your observations, a few points come to mind:

1. **Mosaic Augmentation**: Mosaic data augmentation can sometimes lead to overfitting if the model starts memorizing the augmented images rather than learning generalizable features. You might consider reducing the frequency or intensity of mosaic augmentation if you suspect it's contributing to overfitting.
   
2. **Augmentation Flags**: The `augment=True` flag typically enables a set of augmentation strategies like rotation, scaling, etc., aimed at making your model more robust to variations in the input data. Having `augment=False` would disable these, potentially making your model less adept at generalizing.

To combat overfitting, consider:
- **Adding More Data**: If possible, adding more diverse data can help.
- **Regularization**: Experiment with dropout layers or L2 regularization.
- **Early Stopping**: Monitor validation loss and stop training when it starts to increase, even with the patience parameter set.

Experimenting with these strategies can help mitigate overfitting. Keep iterating on your hyperparameters and augmentation strategies, and you're likely to find a combination that works well.

Lastly, yes, changing the augmentation flag directly impacts the use of augmentations during training. If overfitting persists, tweaking augmentation parameters or exploring additional datasets for training might be beneficial.

Keep up the great work, and all the best with your master's thesis! If you have further questions or hit another roadblock, feel free to ask.

**Comment by glenn-jocher** [comment]
Hello! 👋 You're almost right! The `augment` parameter you're referring to indeed affects inference mode, enabling or disabling augmentations when making predictions. During training, however, data augmentation is controlled by separate parameters defined in the training pipeline and dataset configuration. So, setting `augment=True` or `augment=False` impacts predictions, not the training process. Hope this clears things up! 😊

**Comment by bysdyc** [comment]
> ### 询问前先搜索
> * [x] 我搜索了 YOLOv8[问题](https://github.com/ultralytics/ultralytics/issues)和[讨论](https://github.com/ultralytics/ultralytics/discussions)，没有发现类似的问题。
> 
> ### 问题
> 大家好！ 我正在尝试在 CrowdHuman 数据集上训练 YOLOv8，作为我的硕士论文项目。
> 
> 我做了一些尝试，调整了优化器、学习率和其他超参数，试图找到最佳组合。 与 COCO 数据集上的预训练模型相比，所有指标（P、R、mAP50、mAP50-95）都有所提升。但遗憾的是，在所有尝试中，我注意到验证集上的边界框损失、类别损失和 dfl 损失比训练集上的要差得多，尽管它们趋势相似，这让我怀疑模型存在过拟合。
> 
> **在 COCO 上预训练的 YOLOv8：P = 0.738，R = 0.554，mAP50 = 0.655，mAP50-95 = 0.408**
> 
> 我在这里添加了我在两次最佳尝试中使用的参数和结果：
> 
> `results = model.train(model='yolov8s.pt', data=os.path.join(ROOT_DIR, "config.yaml"), epochs=300, patience = 100, batch=32, imgsz=640, device=[0,1], cache=True, save=True, workers=24, pretrained=True, single_cls=True, optimizer='SGD', deterministic=False, cos_lr=True, lr0=1e-2, lrf=1e-2, augment=False, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=15, translate=0.1, scale=0.5, shear=0.0001, perspective=0.0001, fliplr=0.5, mosaic=1.0, mixup=0.0, copy_paste=0.0, verbose=True, visualize=True)`
> 
> ![results_best2](https://github.com/ultralytics/ultralytics/assets/166723771/2c583523-29fc-41ac-a96a-91ba78ec1801)
> 
> `results = model.train(model='yolov8s.pt', data=os.path.join(ROOT_DIR, "config.yaml"), epochs=300, patience = 100, batch=32, imgsz=640, device=[0,1], cache=True, save=True, workers=24, pretrained=True, single_cls=True, optimizer='SGD', deterministic=False, cos_lr=True, lr0=1e-2, lrf=1e-2, augment=True, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=15, translate=0.1, scale=0.5, shear=0.0001, perspective=0.0001, fliplr=0.5, mosaic=1.0, mixup=0.0, copy_paste=0.0, verbose=True, visualize=True)`
> 
> ![results_best4](https://github.com/ultralytics/ultralytics/assets/166723771/cf1586b8-2a95-4168-a254-1bcd5e3bbc7d)
> 
> 尤其是在最后一次尝试中，我启用了增强功能，`augment=True`并注意到在最后 10 个 epoch 中，当禁用马赛克数据增强时，训练损失会增加到更接近验证数据集损失的值。马赛克数据增强会造成过拟合吗？还是我遗漏了什么？
> 
> 我还有一个疑问：它到底有什么作用`augment=True`？`augment=False`它真的能启用/禁用增强功能吗？
> 
> 非常感谢您的帮助。
> 
> ### 额外的
> _没有回应_

Hello, I also need to use YOLOv8 on the CrowdHuman dataset. Could you please provide a trained model

**Comment by glenn-jocher** [comment]
Thanks—Ultralytics doesn’t share custom CrowdHuman weights via Issues; please train locally (YOLO11 recommended) following the [Train guide](https://docs.ultralytics.com/modes/train/) and tune augments per the [data augmentation guide](https://docs.ultralytics.com/guides/yolo-data-augmentation/). Example setup:

```yaml
# crowdhuman.yaml
path: /path/to/crowdhuman
train: images/train
val: images/val
test: images/test
names: ['person']
```

```bash
yolo detect train model=yolo11m.pt data=crowdhuman.yaml epochs=100 imgsz=640 batch=16 close_mosaic=10
```

```python
from ultralytics import YOLO
model = YOLO('yolo11m.pt')
model.train(data='crowdhuman.yaml', epochs=100, imgsz=640, batch=16, close_mosaic=10)
```

If you must reproduce with YOLOv8, just swap `model=yolov8s.pt`. Since this thread is closed/stale, please open a new Discussion for any follow‑ups so the team can see it.
